File: nak.py
_, plan_count = [int(a) for a in input().split()]
players = [int(string) for string in input().split()]
plans = [[int(i) for i in input().split()] for _ in range(plan_count)]
#for _ in range(plan_count):
#    plans.append(tuple(int(string) for string in input().split()))

for plan in plans:
    relevant_players = players[plan[0]:plan[1]+1]
    print(sorted(relevant_players)[10])

File: nak/nak.cs
using System;
using System.Linq;

namespace nak
{
    class Program
    {
        static void Main(string[] args)
        {
            string[] inputs = Console.ReadLine().Split();
            int planCount = int.Parse(inputs[1]);
            int[] players = Console.ReadLine().Split().Select(int.Parse).ToArray();
            Tuple<int, int>[] plans = new Tuple<int, int>[planCount];

            for (int i = 0; i < planCount; i++)
            {
                int[] planInputs = Console.ReadLine().Split().Select(int.Parse).ToArray();
                plans[i] = new Tuple<int, int>(planInputs[0], planInputs[1]);
            }

            foreach (var plan in plans)
            {
                var sortedPlayers = players.Skip(plan.Item1).Take(plan.Item2 - plan.Item1 + 1).OrderBy(p => p);
                int oldestPlayer = sortedPlayers.ElementAt(10);
                Console.WriteLine(oldestPlayer);
            }
        }
    }
}
